packages/workbox-build/src/lib/utils/modify-url-prefix-transform.js

```diff
@@ -43,7 +42,15 @@ module.exports = (urlString, modifyUrlPrefix) => {
   // a string.
   const modifyRegex = new RegExp(`^(${prefixMatchesStrings})`);
 
-  return urlString.replace(modifyRegex, (match) => {
-    return modifyUrlPrefix[match];
+  return (manifest) => manifest.map((entry) => {
+    if (typeof entry.url !== 'string') {
+      throw new Error(errors['manifest-entry-bad-url']);
+    }
+
+    const modifiedUrl = entry.url.replace(modifyRegex, (match) => {
+      return modifyUrlPrefix[match];
+    });
+
+    return Object.assign({}, entry, {url: modifiedUrl});
```

Comment:
I was doing something stupid in a test that only would work if a copy of the original entry was returned. I've fixed the test and moved to your suggestion.